### PR TITLE
Closes #2249 - Refactor CancelTaskAccTest to use test-api

### LINKED ISSUE #2249: Refactor CancelTaskAccTest to use test-api

### PR DESCRIPTION
**Pull request by jamesrdi**
https://sonarcloud.io/summary/new_code?id=jamesrdi_taskana&branch=TSK-2249

---
Release Notes:

```

```
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] After integration of the PR, I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: lib/taskana-core-test/src/test/java/acceptance/task/complete/CancelTaskAccTest.java [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    defaultObjectReference = defaultTestObjectReference().build();
+
+    // initialise set of tasks so NullStateException will not be thrown when deleting task at the
+    // beginning
+    createTasks();
+  }
+
+  @WithAccessId(user = "admin")
+  @BeforeEach
+  void setUpIndividualTest() throws Exception {
```
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/task/complete/CancelTaskAccTest.java:75]: Please rewrite the test so that it doesn't need this setup bevor each test case.

**Review by ryzheboka — CHANGES_REQUESTED** [review]

File: lib/taskana-core-test/src/test/java/acceptance/task/complete/CancelTaskAccTest.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        TaskBuilder.newTask()
+            .classificationSummary(defaultClassificationSummary)
+            .workbasketSummary(defaultWorkbasketSummary)
+            .state(TaskState.CLAIMED)
+            .primaryObjRef(DefaultTestEntities.defaultTestObjectReference().build())
+            .buildAndStore(taskService);
+
+    taskService.cancelTask(claimedTask.getId());
+
+    Task readTask = taskService.getTask(claimedTask.getId());
```
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/task/complete/CancelTaskAccTest.java:109]: Please remove the line `Task readTask = taskService.getTask(claimedTask.getId());` and read the return value of cancelTask instead

**Review by ryzheboka — CHANGES_REQUESTED** [review]

**Review by ryzheboka — APPROVED** [review]
